cmd/getConfig_test.go

```diff
@@ -147,6 +147,53 @@ func TestApplyContextConditions(t *testing.T) {
 				"dockerImage": "myTestImage:latest",
 			},
 		},
+		{
+			metadata: config.StepData{Spec: config.StepSpec{Containers: []config.Container{
+				{
+					Image: "myDefaultImage:latest",
+					Conditions: []config.Condition{
+						{
+							ConditionRef: "strings-equal",
+							Params: []config.Param{
+								{Name: "param1", Value: "val1"},
+							},
+						},
+					},
+				},
+			}}},
+			conf: config.StepConfig{Config: map[string]interface{}{
+				"param1":      "val1",
+				"dockerImage": "myTestImage:latest",
+			}},
+			expected: map[string]interface{}{
+				"param1":      "val1",
```

Comment:
Writing the tests as an array of "input" and "expected output" structs helps to slim down the tests a lot, but it makes them really hard to understand and follow/validate. I would prefer a number of `t.Run()`s with descriptive titles. I don't think it would blow up the code a lot. From the changed code and the added tests, I don't understand how "a container configuration entry at root level is considered" is achieved, and I also don't understand how the tests are validating behavior when a condition is not met.